### DISCUSSION TITLE
API for Diff for Extension Development.

### DISCUSSION THREAD

**Discussion by sameemqiitm** [microsoft/vscode-discussions#1604] [Extension Development QnA] [answered]
Hey All,
What API/Proposed Api can be used to implement a this diff view specifically inside the inset(inline window)?
Any References/Resources/Examples would really be of great help.
Thanks
![Untitled 3](https://github.com/user-attachments/assets/b67b2119-c5cf-4b22-9973-426d1b570ee8)

**Comment by sameemqiitm** [comment]
Any References???

**Comment by olguzzar** [comment]
Hi @sameemqiitm, check out [this extension sample](https://github.com/microsoft/vscode-extension-samples/tree/main/source-control-sample#populating-changed-files-view) that uses the viewDiff command for showing file diffs. The sample does not show the diff in the inset but could be worth playing around with, if you haven't already, to see if it will give you the results you want.

**Reply by olguzzar** [reply]
Hi @sameemqiitm, did this help? Can you please let us know if you need other assistance?

**Comment by sameemqiitm** [comment]
Hey @olguzzar ,Thanks a lot for the Response :)
Also I had one more Query , I have a menu with some buttons on the webview . On clicking the history button, I want the users history to be displayed . And I want this to be displayed in the webview itself .
hat would be the best way to achieve this?
![logi ai menu](https://github.com/user-attachments/assets/c14e6e74-2f69-4613-a6b2-83827b2f550e)

**Reply by robconery** [reply, marked as answer]
Would you mind popping this into a new Discussion so we can track better?
